src/OpenSage.Game/Graphics/Animation/W3DAnimation.cs

```diff
@@ -12,6 +12,12 @@ internal static W3DAnimation FromW3dFile(W3dFile w3dFile)
 
             var animations = new W3DAnimation[w3dAnimations.Count + w3dCompressedAnimations.Count];
 
+            if (animations.Length == 0)
```

Comment:
sometimes objects reference themselfes as animations in animation states (e.g. gbforbidpool from bfme2) thus the referenced w3d file does not contain any animation data at all.